stonesoup/types/time.py

```diff
@@ -21,33 +22,278 @@ class TimeRange(Type):
     True
     """
 
-    start_timestamp: datetime.datetime = Property(doc="Start of the time range")
-    end_timestamp: datetime.datetime = Property(doc="End of the time range")
+    start: datetime.datetime = Property(doc="Start of the time range")
+    end: datetime.datetime = Property(doc="End of the time range")
 
-    def __init__(self, start_timestamp, end_timestamp, *args, **kwargs):
-        if end_timestamp < start_timestamp:
-            raise ValueError("start_timestamp must be before end_timestamp")
-        super().__init__(start_timestamp, end_timestamp, *args, **kwargs)
+    @property
+    def start_timestamp(self):
+        return self.start
+
+    @property
+    def end_timestamp(self):
+        return self.end
 
     @property
     def duration(self):
         """Duration of the time range"""
 
-        return self.end_timestamp - self.start_timestamp
+        return self.length
+
+    @property
+    def key_times(self):
+        """Times the TimeRange begins and ends"""
+        return [self.start, self.end]
 
-    def __contains__(self, timestamp):
+    def __contains__(self, time):
         """Checks if timestamp is within range
 
         Parameters
         ----------
-        timestamp : datetime.datetime
-            Time stamp to check if within range
+        time : Union[datetime.datetime, TimeRange]
+            Time stamp or range to check if within range
 
         Returns
         -------
         bool
-            `True` if timestamp within :attr:`start_timestamp` and
-            :attr:`end_timestamp` (inclusive)
+            `True` if timestamp within :attr:`start` and
+            :attr:`end` (inclusive)
+        """
+        if isinstance(time, datetime.datetime):
+            return self.start <= time <= self.end
+        else:
+            return super().__contains__(time)
+
+    def __eq__(self, other):
+        return isinstance(other, TimeRange) and super().__eq__(other)
+
+    def minus(self, time_range):
```

Comment:
Shouldn't this be same as or override `__sub__`?

Reply:
Yeah, they're similar, so renaming.  `time_range` can be either a `TimeRange` or a `CompoundTimeRange` interval though here, as opposed to limiting it to being the same type.